lib/spack/spack/cmd/buildcache.py

```diff
@@ -1004,10 +1005,17 @@ def manifest_copy(manifest_file_list):
                 # Last duplicate hash wins
                 deduped_manifest[spec_hash] = copy_list
 
+    build_cache_dir = bindist.build_cache_relative_path()
     for spec_hash, copy_list in deduped_manifest.items():
         for copy_file in copy_list:
-            tty.debug("copying {0} to {1}".format(copy_file["src"], copy_file["dest"]))
-            copy_buildcache_file(copy_file["src"], copy_file["dest"])
+            dest = copy_file["dest"]
+            if dest_mirror:
+                src_relative_path = os.path.join(
```

Comment:
Is this block exercised anywhere?  It seems to be the key change in this PR, but I don't think it's tested.

Reply:
right. I need to figure out how to do that still. I just wrote this quickly and did a local test to make sure it worked, but that was it.